tests/jerry/array-prototype-slice.js

```diff
@@ -151,3 +151,33 @@ try {
   assert (e.message === "foo");
   assert (e instanceof ReferenceError);
 }
+
+// Remove the buffer
+var array = [1, 2, 3, 4, 5];
+var value = array.slice(4, {
+    valueOf: function() {
+        array.length = 0;
+    }
+})
+
+print(JSON.stringify(value) === '[]')
```

Comment:
This should be an assert. And just to make sure, we should avoid the JSON.stringify calls in the future ( we could use an helper method to do the comparison like in: https://github.com/jerryscript-project/jerryscript/commit/908240ba6223fb54e72c2ab85c3765db3492a968#diff-2b73935d70c05987656fb33f1150f9d1R19 ). The same applies to the parts below.